packages/luna/src/components/Chat/ChatMessage/ChatMessage.tsx

```diff
@@ -35,67 +85,64 @@ export const ChatMessage: React.FC<{
+              setOpenMenu(open);
+            }}
+          >
+            <div className="tw-w-6 tw-h-6 tw-flex tw-justify-center tw-items-center">
+              <More className="[&>*]:tw-fill-natural-800 dark:[&>*]:tw-fill-natural-50" />
+            </div>
+          </Menu>
+        </div>
+      ) : null}
+
       <div
-        data-show={showMenu}
         className={cn(
-          "tw-opacity-0 data-[show=true]:tw-opacity-100 tw-transition-opacity tw-duration-200",
-          viewOnly && "tw-hidden"
+          "tw-rounded-[15px] tw-relative tw-px-6 tw-py-3",
+          "tw-flex tw-items-center tw-gap-2 tw-justify-start",
+          {
+            "tw-bg-brand-100 dark:tw-bg-brand-100 tw-rounded-tl-none": !owner,
+            "tw-bg-brand-700 dark:tw-bg-brand-400 tw-rounded-tr-none": owner,
+            "tw-bg-destructive-700": error,
+          }
         )}
       >
-        <Menu
-          actions={[
-            {
-              label: intl("chat.message.edit"),
-              onClick: editMessage,
-              icon: <MessageEdit />,
-            },
-            {
-              label: intl("chat.message.delete"),
-              onClick: deleteMessage,
-              icon: <Trash />,
-            },
-          ]}
-          open={openMenu}
-          setOpen={(open: boolean) => {
-            if (!open) setShowMenu(false);
-            setOpenMenu(open);
-          }}
-        >
-          <div className="tw-w-6 tw-h-6 tw-flex tw-justify-center tw-items-center">
-            <More className="[&>*]:tw-fill-natural-800 dark:[&>*]:tw-fill-natural-50" />
-          </div>
-        </Menu>
-      </div>
-      <div
-        className={cn("tw-rounded-[15px] tw-relative tw-px-6 tw-py-3", {
-          "tw-bg-brand-100 dark:tw-bg-brand-100": !owner,
-          "tw-bg-brand-700 dark:tw-bg-brand-400": owner,
-        })}
-      >
-        {!owner ? (
-          <div
-            className={cn(
-              "tw-absolute -tw-top-[10px] tw-left-[0px] tw-w-0 tw-h-0",
-              "tw-rounded-t-full tw-rounded-r-full",
-              "tw-border-r-[40px] tw-border-b-[22px]",
-              "tw-border-r-transparent tw-border-b-brand-100 dark:tw-border-b-brand-100"
-            )}
-          />
-        ) : (
+        <MessageCap
+          className={cn("-tw-top-1 tw-absolute", {
+            "-tw-right-0 [&>*]:tw-fill-brand-700": owner,
+            "-tw-right-0 [&>*]:tw-fill-destructive-700": error,
+            "-tw-left-0 tw-scale-x-[-1] [&>*]:tw-fill-brand-100": !owner,
+          })}
+        />
+        {error ? (
           <div
             className={cn(
-              "tw-absolute -tw-top-[10px] tw-right-[0px] tw-w-0 tw-h-0",
-              "tw-rounded-t-full tw-rounded-l-full",
-              "tw-border-l-[40px]  tw-border-b-[22px]",
-              "tw-border-l-transparent tw-border-b-brand-700 dark:tw-border-b-brand-400"
+              "tw-min-w-6 tw-min-h-6 tw-rounded-full tw-bg-destructive-500 tw-shadow-alert-circle",
```

Comment:
No need to add `tw-min-*` you can just use `tw-w-6 tw-h-6`. In case you want to fix the issue shrinking the image, you can use `tw-shrink-0`